packages/framework/src/Model/Order/OrderData.php

```diff
@@ -331,4 +324,23 @@ public function getTotalPriceWithoutDiscountTransportAndPayment(): Price
             ->subtract($this->totalPricesByItemType[OrderItemTypeEnum::TYPE_PAYMENT])
             ->subtract($this->totalPricesByItemType[OrderItemTypeEnum::TYPE_DISCOUNT]);
     }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData[]
+     */
+    public function getItemsWithoutTransportAndPayment(): array
+    {
+        return array_filter(
+            $this->items,
+            fn (OrderItemData $item) => !in_array($item->type, [OrderItemTypeEnum::TYPE_TRANSPORT, OrderItemTypeEnum::TYPE_PAYMENT], true),
+        );
+    }
+
+    /**
+     * @param $items
+     */
+    public function setItemsWithoutTransportAndPayment($items): void
```

Comment:
this is never used :thinking:

Reply:
This is used by form type `OrderItemsType`

Reply:
let's discuss it please, it opens a few questions :smile: 

- should not be there a typehint for $items param? (+ the annotation is missing a type)
- is it ok that it sets the `items` property then? if the form field was named just `items`, we would not need the setter, right? Or, maybe even better, could not we rename the `items` property to `itemsWithoutTransportAndPayment`? :thinking:

Reply:
added comment to describe the reason and type to property and annotation 👍